### PR TITLE
Datastore alignment

### PR DESCRIPTION
**Pull request by m00sey**
This removes support for couchdb and postgres as Canis datastores.
Separates Aries's storage and supports their couchdb and MySQL options.

To avoid disrupting development #60 should be completed and this work updated to add mongo as an option to the Aries storage

### CODE REVIEW

File: cmd/issuer/main.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ func createAriesContext() {
 	wsinbound := "0.0.0.0:2001"
 
 	ar, err := aries.New(
-		aries.WithStoreProvider(store.NewProvider("mongodb://172.17.0.1:27017", "issuer")),
+		aries.WithStoreProvider(mongodbstore.NewProvider("mongodb://172.17.0.1:27017", mongodbstore.WithDBPrefix("aries"))),
```
Comment (pfeairheller) [inline cmd/issuer/main.go:99]: If you change all these to "aries" as the prefix, they will collide with you run them at the same time locally.

File: cmd/subject/main.go [hunk 2]
```diff
@@ -150,7 +149,7 @@ func createAriesContext() {
 	//amqpInbound, err := amqp.NewInbound(wsinbound, "ws://0.0.0.0:3001", "", "")
 
 	ar, err := aries.New(
-		aries.WithStoreProvider(store.NewProvider("mongodb://172.17.0.1:27017", "subject")),
+		//aries.WithStoreProvider(store.NewProvider("mongodb://172.17.0.1:27017", "subject")),
```
Comment (pfeairheller) [inline cmd/subject/main.go:152]: I assume you just forgot to add the new one back in?

File: cmd/verifier/main.go [hunk 3]
```diff
@@ -108,7 +108,7 @@ func createAriesContext() {
 	wsinbound := "0.0.0.0:4001"
 
 	ar, err := aries.New(
-		aries.WithStoreProvider(store.NewProvider("mongodb://172.17.0.1:27017", "verifier")),
+		aries.WithStoreProvider(mongodbstore.NewProvider("mongodb://172.17.0.1:27017", mongodbstore.WithDBPrefix("aries"))),
```
Comment (pfeairheller) [inline cmd/verifier/main.go:111]: Same comment about "aries"

File: pkg/credential/client.go [hunk 4]
```diff
@@ -24,7 +24,7 @@ type credKeys struct {
 
 func NewClient() (*Client, error) {
 	//provider := conf.GetAriesContext()
-	//store, err := provider.StorageProvider().OpenStore("cred-defs")
+	//store, err := provider.StorageProvider().Open("cred-defs")
```
Comment (pfeairheller) [inline pkg/credential/client.go:27]: I very much appreciate you changing commented code!

File: pkg/didcomm/loadbalancer/cmd/root.go [hunk 5]
```diff
@@ -75,27 +74,20 @@ func initConfig() {
...
-	store, err := sp.OpenStore("canis")
-	if err != nil {
-		log.Fatalln("unable to open store")
-	}
-	asp, err := store.GetAriesProvider()
-
 	prov = &Provider{
 		vp:                   vp,
-		ariesStorageProvider: asp,
+		ariesStorageProvider: ls,
```
Comment (pfeairheller) [inline pkg/didcomm/loadbalancer/cmd/root.go:90]: Every time I see `ls` I want to comment that you are missing `-al`.  So there, I did it once.

File: pkg/framework/context/coverage.txt [hunk 6]
```diff
@@ -0,0 +1,91 @@
+mode: atomic
```
Comment (pfeairheller) [inline pkg/framework/context/coverage.txt:1]: We don't want these in the codebase, do we?

File: pkg/datastore/mongodb/coverage.txt [hunk 7]
```diff
@@ -0,0 +1,74 @@
+mode: atomic
```
Comment (pfeairheller) [inline pkg/datastore/mongodb/coverage.txt:1]: We don't want these do we?

File: pkg/framework/coverage.txt [hunk 8]
```diff
@@ -0,0 +1,25 @@
+mode: atomic
```
Comment (pfeairheller) [inline pkg/framework/coverage.txt:1]: Samsies

File: pkg/steward/coverage.txt [hunk 9]
```diff
@@ -0,0 +1,140 @@
+mode: atomic
```
Comment (pfeairheller) [inline pkg/steward/coverage.txt:1]: Ditto

**Review by pfeairheller — CHANGES_REQUESTED** [review]
The datastore names for the `cmd/[issuer|verifier|subject]` are only true because those are sample apps that I wrote to test out the entire process before moving it into canis.  The `subject` is the only one we are likely to keep around because it represents a prover (or subject) which we won't be representing in canis (for now...)

File: .gitignore [hunk 10]
```diff
@@ -1,3 +1,3 @@
 /.idea/
 /bin/
-coverage.txt
```
Comment (pfeairheller) [inline .gitignore:3]: Probably need this back in

File: cmd/issuer/main.go (hunk 1 above)
Comment (m00sey) [reply cmd/issuer/main.go:99]: I believe I meant to prefix with `issuer` etc. This comes back to the change I made to aries-storage-mongo to follow the 1 database / 1 collection approach... We don't have to stick with it.

File: cmd/subject/main.go (hunk 2 above)
Comment (m00sey) [reply cmd/subject/main.go:152]: yes

File: pkg/credential/client.go (hunk 4 above)
Comment (m00sey) [reply pkg/credential/client.go:27]: rektw

**Review by pfeairheller — APPROVED** [review]

**Pull request merged**